bin/BuildPackages.sh

```diff
@@ -315,7 +316,10 @@ do
     fi
   else
     # wait for this package to finish building
-    wait
+    if ! wait $BUILD_PID && [[ $STRICT = yes ]]
```

Comment:
So my understanding of shell job control is fairly limited, so the following is possibly "obvious" if one knows this stuff, but it isn't to me: I don't understand, why do we need `$BUILD_PID`? Why doesn't this work:
```suggestion
    if ! wait && [[ $STRICT = yes ]]
```

Reply:
`wait` always returns `0` if called without a process ID as argument:
```
$ help wait
wait: wait [n]
    Wait for the specified process and report its termination status.  If
    N is not given, all currently active child processes are waited for,
    and the return code is zero. [...]
```

Reply:
Aaaah of course!

so could we adapt https://stackoverflow.com/questions/356100/how-to-wait-in-bash-for-several-subprocesses-to-finish-and-return-exit-code-0#356154 ?

Reply:
I'm not sure – this seems complicated, as @ChrisJefferson has found.

`BuildPackages.sh --parallel` currently runs only <= 4 jobs in the background, and (by default) uses `wait -n` (which is not supported on my version of bash, see #4291) to wait for _any one_ of those jobs to finish, before despatching one more job. (`wait -n` _does_ pass on the exit code of the job that finished... but surely you'd miss checking some exit codes, if you relied on check them this way...)

This is as opposed to not using `wait -n` but just normal `wait` and waiting for _all 4 jobs_ to terminate before spawning any more, which on first glance seems to be what would be required of that Stack Overflow answer.

Alternatively, we could run all jobs simultaneously in the background without a limit, but I presume this is undesirable (it is what @ChrisJefferson initially implemented, before moving to a limit of 4).